### PR TITLE
feat(ui)!: add Hungarian localization

### PR DESCRIPTION
**Pull request by cselti**
## Description

Add Hungarian translation to flutterfire_ui.

## Related Issues

#9052 
#9054 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/flutterfire_ui/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.4.2+3
```
Comment (russellwheatley) [inline packages/flutterfire_ui/CHANGELOG.md:1]: please revert CHANGELOG changes as it is updated automatically.

File: packages/flutterfire_ui/pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: flutterfire_ui
```
Comment (russellwheatley) [inline packages/flutterfire_ui/pubspec.yaml:1]: Please revert changes to pubspec file as it is updated automatically.

**Review by russellwheatley — CHANGES_REQUESTED** [review]
Good work, @cselti! Just a few things:

1. Make changes as request.
2. Sign the CLA agreement.
3. Do you know any other Hungarian/Hungarian speaking developers on Github who could possibly proofread your translations (just in case of mistakes, we all make them 😄)?

**Comment by cselti** [comment]
@russellwheatley 

> 1. Make changes as request.

Sorry for the stupid question, but how can I do this? I should re-create the PR according to the Contributing guide?

**Comment by russellwheatley** [comment]
> @russellwheatley
> 
> > 1. Make changes as request.
> 
> Sorry for the stupid question, but how can I do this? I should re-create the PR according to the Contributing guide?

You've already made the changes 😄 (removed changes to pubspec and CHANGELOG). Just need a proof reader to check the Hungarian translation to approve, please.

**Comment by cselti** [comment]
@AndrasTimar and @norbert-sziraczki can approve my Hungarian translation.

File: packages/flutterfire_ui/lib/src/i10n/lang/hu.dart [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    this.signInHintText = 'Már van fiókod?',
+    this.signOutButtonText = 'Kijelentkezés',
+    this.phoneInputLabel = 'Telefonszám',
+    this.phoneNumberInvalidErrorText = 'Érvénytelen telefonszám',
+    this.phoneNumberIsRequiredErrorText = 'Telefonszám megadása kötelező',
+    this.profile = 'Profil',
+    this.name = 'Név',
+    this.deleteAccount = 'Fiók törlése',
+    this.passwordIsRequiredErrorText = 'Jelszó megadása kötelező',
+    this.confirmPasswordIsRequiredErrorText = 'Jelszó megerősítése',
```
Comment (AndrasTimar) [inline packages/flutterfire_ui/lib/src/i10n/lang/hu.dart:215]: In my opinion this should be:
"Erősítd meg a jelszavad"

**Comment by GyurkovicsFeri** [comment]
> @AndrasTimar and @norbert-sziraczki and @GyurkovicsFeri can approve my Hungarian translation.

I've read it through and compared it with the English version and for me, as a native speaker, it looks and sounds good.

File: packages/flutterfire_ui/lib/src/i10n/lang/hu.dart [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    this.confirmPasswordDoesNotMatchErrorText = 'Jelszavak nem egyeznek',
+    this.confirmPasswordInputLabel = 'Jelszó ismét',
+    this.forgotPasswordButtonLabel = 'Efelejtett jelszó?',
+    this.forgotPasswordViewTitle = 'Elfelejtett jelszó',
+    this.resetPasswordButtonLabel = 'Jelszó visszaállítása',
+    this.verifyItsYouText = 'Igazold magad',
+    this.differentMethodsSignInTitleText =
+        'Válassz az alábbi bejelentkezés lehetőségek közül',
```
Comment (AndrasTimar) [inline packages/flutterfire_ui/lib/src/i10n/lang/hu.dart:223]: I suggest 
'Válassz az alábbi bejelentkezés**i** lehetőségek közül'

File: packages/flutterfire_ui/lib/src/i10n/lang/hu.dart (hunk 3 above)
Comment (GyurkovicsFeri) [reply packages/flutterfire_ui/lib/src/i10n/lang/hu.dart:215]: Yes, this sounds better for me, too. Although the `Erősítsd meg a jelszavad:` is a bit informal for me. The original is not the best in it's context, but totally natural.

File: packages/flutterfire_ui/lib/src/i10n/lang/hu.dart [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+    this.findProviderForEmailTitleText = 'Add meg az email címed a folytatáshoz',
+    this.continueText = 'Folytatás',
+    this.countryCode = 'Országkód',
+    this.codeRequiredErrorText = 'Országkód megadása kötelező',
+    this.invalidCountryCode = 'Érvénytelen országkód',
+    this.chooseACountry = 'Válassz országot',
+    this.enableMoreSignInMethods = 'Több bejelentkezési módszer engedélyezése',
+    this.signInMethods = 'Bejelentkezési módszerek',
```
Comment (AndrasTimar) [inline packages/flutterfire_ui/lib/src/i10n/lang/hu.dart:231]: i'd suggest using 'Bejelentkezési lehetőségek' to stay consistent with the above

File: packages/flutterfire_ui/lib/src/i10n/lang/hu.dart [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+    this.verifyingPhoneNumberViewTitle = 'Add meg az SMS kódot',
+    this.unknownError = 'Ismeretlen hiba történt',
+    this.smsAutoresolutionFailedError =
+        'Nem sikerült az SMS kódot kinyerni. Kérlek add meg a kódot manuálisan',
+    this.smsCodeSentText = 'SMS kód elküldve',
+    this.sendingSMSCodeText = 'SMS kód küldése...',
+    this.verifyingSMSCodeText = 'SMS kód ellenőrzése...',
+    this.enterSMSCodeText = 'Add meg az SMS kódot',
+    this.emailIsRequiredErrorText = 'Email cím megadása kötelező',
+    this.isNotAValidEmailErrorText = 'Helyes email címet adj meg',
```
Comment (GyurkovicsFeri) [inline packages/flutterfire_ui/lib/src/i10n/lang/hu.dart:196]: For me, this is more formal and polite if we tell the fact, that the email is not valid and does not command the user.

```suggestion
    this.isNotAValidEmailErrorText = 'Helytelen email cím',
```

File: packages/flutterfire_ui/lib/src/i10n/lang/hu.dart [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+    this.signOutButtonText = 'Kijelentkezés',
+    this.phoneInputLabel = 'Telefonszám',
+    this.phoneNumberInvalidErrorText = 'Érvénytelen telefonszám',
+    this.phoneNumberIsRequiredErrorText = 'Telefonszám megadása kötelező',
+    this.profile = 'Profil',
+    this.name = 'Név',
+    this.deleteAccount = 'Fiók törlése',
+    this.passwordIsRequiredErrorText = 'Jelszó megadása kötelező',
+    this.confirmPasswordIsRequiredErrorText = 'Jelszó megerősítése',
+    this.confirmPasswordDoesNotMatchErrorText = 'Jelszavak nem egyeznek',
```
Comment (GyurkovicsFeri) [inline packages/flutterfire_ui/lib/src/i10n/lang/hu.dart:216]: This sounds better for me.

```suggestion
    this.confirmPasswordDoesNotMatchErrorText = 'A megadott jelszavak nem egyeznek',
```

**Pull request closed without merging**